contributed_definitions/nyaml/NXpiezo_config_spm.yaml

```diff
@@ -0,0 +1,59 @@
+category: base
+doc: |
+  A base class to describe piezo settings for scanning probe microscopy.
+
+  The material of the Piezoelectric scanner is composed of polycrystalline solids and 
+  attached at the head of the tip. The material is highly sensitive to applied voltage. 
+# No need to create this ´base class rather we can define in spm application definition.
+NXpiezo_config_spm(NXobject):
+  piezo_material(NXpiezoelectric_material):
+    doc: |
+      The material description and properties of the piezoelectric scanner materials.
+  active_calib:
+    doc: |
+      The name of the calibration type (e.g. LHe).
+  calib_N(NX_NUMBER):
+    exists: optional
+    doc: |
+      N denotes X or Y or Z. There are three directions X, Y, and Z for calibration, along with
+      three available controls: Calibration (m/V), Range (m), and HV gain each. Only
```

Comment:
Wouldn't you want to store these parameters within the calibration?

Reply:
The parameters are there exactly after this field.